PWGEM/PhotonMeson/Utils/gammaSelectionCuts.h

```diff
@@ -0,0 +1,158 @@
+std::vector<float> EMC_Eoverp = {1.75f};
+} // namespace emccuts
+void gatherCutsEMC(float minTime, float maxTime, float minM02, float maxM02, float minE, int minNCell, std::vector<float> TM_Eta, std::vector<float> TM_Phi, float Eoverp)
+{
+  // insert the configurable values in first position
+  emccuts::EMC_minTime.insert(emccuts::EMC_minTime.begin(), minTime);
+  emccuts::EMC_maxTime.insert(emccuts::EMC_maxTime.begin(), maxTime);
+  emccuts::EMC_minM02.insert(emccuts::EMC_minM02.begin(), minM02);
+  emccuts::EMC_maxM02.insert(emccuts::EMC_maxM02.begin(), maxM02);
+  emccuts::EMC_minE.insert(emccuts::EMC_minE.begin(), minE);
+  emccuts::EMC_minNCell.insert(emccuts::EMC_minNCell.begin(), minNCell);
+  emccuts::EMC_TM_Eta.insert(emccuts::EMC_TM_Eta.begin(), TM_Eta);
+  emccuts::EMC_TM_Phi.insert(emccuts::EMC_TM_Phi.begin(), TM_Phi);
+  emccuts::EMC_Eoverp.insert(emccuts::EMC_Eoverp.begin(), Eoverp);
+
+  // fill up the rest of the vectors to size 64 to ensure no crashes can happen
+  emccuts::EMC_minTime.resize(64, 0);
+  emccuts::EMC_maxTime.resize(64, 0);
+  emccuts::EMC_minM02.resize(64, 0);
+  emccuts::EMC_maxM02.resize(64, 0);
+  emccuts::EMC_minE.resize(64, 0);
+  emccuts::EMC_minNCell.resize(64, 0);
+  emccuts::EMC_TM_Eta.resize(64, {0, 0, 0});
+  emccuts::EMC_TM_Phi.resize(64, {0, 0, 0});
+  emccuts::EMC_Eoverp.resize(64, 0);
+}
+
+uint64_t doTimeCutEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, HistogramRegistry registry)
+{
+  uint64_t cut_return = 0;
+  if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) {                                                             // check if current cut should be applied
+    if (cluster.time() <= emccuts::EMC_maxTime.at(iCut) && cluster.time() >= emccuts::EMC_minTime.at(iCut)) { // check cut itself
+      cut_return |= (1 << iCut);                                                                              // set bit of current cut to 1 for passing the cut
+    } else {
+      registry.fill(HIST("hCaloCuts_EMC"), 1, iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doM02CutEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, HistogramRegistry registry)
+{
+  uint64_t cut_return = 0;
+  if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) {                                                         // check if current cut should be applied
+    if (cluster.m02() <= emccuts::EMC_maxM02.at(iCut) && cluster.m02() >= emccuts::EMC_minM02.at(iCut)) { // check cut itself
+      cut_return |= (1 << iCut);                                                                          // set bit of current cut to 1 for passing the cut
+    } else {
+      registry.fill(HIST("hCaloCuts_EMC"), 2, iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doMinECutEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, HistogramRegistry registry)
+{
+  uint64_t cut_return = 0;
+  if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) {        // check if current cut should be applied
```

Comment:
in principle this condition has already been checked in the processRec of gammaSelection, right?

Reply:
Yes, I left it for safety reasons. Could be removed I think.

Reply:
No, see comment below regarding the propagation of the cut_return value instead of the cutbit variable after the timing cut.